File: src/handlers/init-handler.js
import { Wallet } from "../class/Wallet.js";
import { input } from "../utils/input.js";
import { validateString } from "../utils/validate-string.js";
import { errorhandler } from "./error-handler.js";

export const initHandler = async (setWallet) =>
  await errorhandler(async () => {
    const color = await input("Color of the wallet => ");
    validateString(color);
    const mark = await input("Mark of the wallet => ");
    validateString(mark);
    setWallet(new Wallet(color, mark));
  });

File: src/handlers/init_handler.dart
import '../class/wallet.dart';
import '../utils/input.dart';
import '../utils/validate_string.dart';
import 'error_handler.dart';

Future<void> initHandler(void Function(Wallet) setWallet) async {
  await errorHandler(() async {
    final color = input('Color of the wallet => ');
    validateString(color);
    final mark = input('Mark of the wallet => ');
    validateString(mark);
    setWallet(Wallet(color, mark));
  });
}
